src/act/node/canister_method/query_method.rs

```diff
@@ -51,6 +43,26 @@ impl QueryMethod {
     }
 }
 
+impl Deref for QueryMethod {
+    type Target = QueryOrUpdateDefinition;
+
+    fn deref(&self) -> &Self::Target {
+        &self.definition
+    }
+}
+
+impl HasParams for QueryMethod {
```

Comment:
Can you move the HasParams and HasReturnValue impls bellow the Proclaim impl to be consistent with other files' organizations?